app/src/components/keyCloakServiceClientMgr.js

```diff
@@ -169,17 +169,23 @@ class KeyCloakServiceClientManager {
     }
   }
 
-  async fetchAllClients() {
-    log.info('KeyCloakServiceClientManager.fetchAllClients');
+  async fetchAllClients(kcEnv) {
+
+    log.info('KeyCloakServiceClientManager.fetchAllClients in ' + kcEnv);
 
     //get all service clients
-    // NOTE: this seems to return all the service clinets but shows a ERR 400 in the console for all the methods in the reamlAdminSvc class
-    const clients = await this.svc.getClients();
+    let clients = await this.svc.getClients();
+
+    // remove those that dont match '*_SERVICE_CLIENT';
+    const regex = '.*_SERVICE_CLIENT$';
+    clients = clients.filter(cl => cl.clientId.match(regex));
 
-    // get details for each service client
-    const clientsWithDetails = clients.map(sc => this.makeClientDetails(sc));
+    // add the keycloak environment to each service client object in the array
+    clients.forEach(function (cl) {
+      cl.realm = kcEnv;
+    });
 
-    return await Promise.all(clientsWithDetails);
+    return clients;
```

Comment:
To extend on the above comments, you can choose to do array filter chaining:
```
const clients = await this.svc.getClients();
return clients
    .filter(cl => cl.clientId.match(/.*_SERVICE_CLIENT$/))
    .map(cl => { ...cl, realm: kcEnv });
```
This should be functionally equivalent.